lib/model/channel.dart

```diff
@@ -199,6 +199,51 @@ class ChannelStoreImpl with ChannelStore {
           streamsByName.remove(stream.name);
           subscriptions.remove(stream.streamId);
         }
+
+      case ChannelUpdateEvent():
+        final ChannelUpdateEvent(:streamId, name: String streamName) = event;
+        assert(identical(streams[streamId], streamsByName[streamName]));
+        assert(subscriptions[streamId] == null
+          || identical(subscriptions[streamId], streams[streamId]));
+
+        final channel = streams[streamId];
+        if (channel == null) return;
+
+        if (event.renderedDescription != null) {
+          channel.renderedDescription = event.renderedDescription!;
+        }
+        if (event.historyPublicToSubscribers != null) {
+          channel.historyPublicToSubscribers = event.historyPublicToSubscribers!;
+        }
+        if (event.isWebPublic != null) {
+          channel.isWebPublic = event.isWebPublic!;
+        }
+
+        if (event.property == null) {
+          // unrecognized property; do nothing
+          return;
+        }
+        switch (event.property!) {
+          case ChannelPropertyName.name:
+            channel.name = event.value as String;
+            streamsByName.remove(streamName);
+            streamsByName[channel.name] = channel;
+          case ChannelPropertyName.description:
+            channel.description = event.value as String;
+          case ChannelPropertyName.firstMessageId:
+            channel.firstMessageId = event.value as int?;
+          case ChannelPropertyName.inviteOnly:
+            channel.inviteOnly = event.value as bool;
+          case ChannelPropertyName.messageRetentionDays:
+            channel.messageRetentionDays = event.value as int?;
+          case ChannelPropertyName.channelPostPolicy:
+            channel.channelPostPolicy = event.value as ChannelPostPolicy;
+          case ChannelPropertyName.canRemoveSubscribersGroup:
+          case ChannelPropertyName.canRemoveSubscribersGroupId:
+            channel.canRemoveSubscribersGroup = event.value as int?;
```

Comment:
I think it's probably OK to accept null for this one, but from reading the API docs (including the register-queue doc), I'm not sure it's ever expected, and what it would mean. The field on `ZulipStream` is nullable, but I think that's only for backwards compatibility with older servers that don't have the field:

```dart
  // TODO(server-6): `canRemoveSubscribersGroupId` added in FL 142
  // TODO(server-8): in FL 197 renamed to `canRemoveSubscribersGroup`
  @JsonKey(readValue: _readCanRemoveSubscribersGroup)
  int? canRemoveSubscribersGroup;
```

And if we're in this `case`, then surely we're not on one of those older servers.

So, probably nothing to do here, and no harm in being a little more permissive with `int?` instead of `int`, I think, as long as the field on `ZulipStream` remains nullable.